server/src/test/java/org/elasticsearch/action/ingest/SimulateProcessorResultTests.java

```diff
@@ -72,4 +63,64 @@ public void testSerialization() throws IOException {
             assertThat(e.getMessage(), equalTo("test"));
         }
     }
+
+    protected static SimulateProcessorResult createTestInstance(boolean isSuccessful,
+                                                                boolean isIgnoredException,
+                                                                boolean withByteArray) {
+        String processorTag = randomAlphaOfLengthBetween(1, 10);
+        SimulateProcessorResult simulateProcessorResult;
+        if (isSuccessful) {
+            IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), withByteArray);
+            if (isIgnoredException) {
+                simulateProcessorResult = new SimulateProcessorResult(processorTag, ingestDocument, new IllegalArgumentException("test"));
+            } else {
+                simulateProcessorResult = new SimulateProcessorResult(processorTag, ingestDocument);
+            }
+        } else {
+            simulateProcessorResult = new SimulateProcessorResult(processorTag, new IllegalArgumentException("test"));
+        }
+        return simulateProcessorResult;
+    }
+
+    @Override
+    protected SimulateProcessorResult createTestInstance() {
+        boolean isSuccessful = randomBoolean();
+        boolean isIgnoredException = randomBoolean();
+        return createTestInstance(isSuccessful, isIgnoredException, false);
+    }
+
+    @Override
+    protected SimulateProcessorResult doParseInstance(XContentParser parser) {
+        return SimulateProcessorResult.fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
same comment as above, let's return false here only if we can't insert random fields ever in any level of the object